src/Lykke.Job.TransactionHandler/Handlers/EthereumCoreCommandHandler.cs

```diff
@@ -0,0 +1,468 @@
+        private readonly IAssetsService _assetsService;
+        private readonly IEthererumPendingActionsRepository _ethererumPendingActionsRepository;
+        private readonly IDeduplicator _deduplicator;
+        private readonly IExchangeOperationsServiceClient _exchangeOperationsServiceClient;
+        private readonly IClientCommentsRepository _clientCommentsRepository;
+
+        public EthereumCoreCommandHandler(AppSettings.RabbitMqSettings config, ILog log,
+            IMatchingEngineClient matchingEngineClient,
+            ICashOperationsRepositoryClient cashOperationsRepositoryClient,
+            IClientAccountClient clientAccountClient,
+            ISrvEmailsFacade srvEmailsFacade,
+            IBcnClientCredentialsRepository bcnClientCredentialsRepository,
+            IPaymentTransactionsRepository paymentTransactionsRepository,
+            IWalletCredentialsRepository walletCredentialsRepository,
+            ITradeOperationsRepositoryClient clientTradesRepositoryClient,
+            IEthereumTransactionRequestRepository ethereumTransactionRequestRepository,
+            ITransferOperationsRepositoryClient transferEventsRepositoryClient,
+            IAssetsServiceWithCache assetsServiceWithCache,
+            ITransactionService transactionService,
+            IAssetsService assetsService,
+            IEthererumPendingActionsRepository ethererumPendingActionsRepository,
+            IExchangeOperationsServiceClient exchangeOperationsServiceClient,
+            IClientCommentsRepository clientCommentsRepository)
+        {
+            _log = log.CreateComponentScope(nameof(EthereumCoreCommandHandler));
+            _matchingEngineClient = matchingEngineClient;
+            _cashOperationsRepositoryClient = cashOperationsRepositoryClient;
+            _clientAccountClient = clientAccountClient;
+            _srvEmailsFacade = srvEmailsFacade;
+            _bcnClientCredentialsRepository = bcnClientCredentialsRepository;
+            _paymentTransactionsRepository = paymentTransactionsRepository;
+            _walletCredentialsRepository = walletCredentialsRepository;
+            _clientTradesRepositoryClient = clientTradesRepositoryClient;
+            _ethereumTransactionRequestRepository = ethereumTransactionRequestRepository;
+            _assetsServiceWithCache = assetsServiceWithCache;
+            _transferEventsRepositoryClient = transferEventsRepositoryClient;
+            _transactionService = transactionService;
+            _assetsService = assetsService;
+            _ethererumPendingActionsRepository = ethererumPendingActionsRepository;
+            _exchangeOperationsServiceClient = exchangeOperationsServiceClient;
+            _clientCommentsRepository = clientCommentsRepository;
+        }
+
+        #region CommanHandling
+
+        public async Task<CommandHandlingResult> Handle(ProcessHotWalletEventCommand command, IEventPublisher eventPublisher)
```

Comment:
I remember that we have discussed about ProcessHotWalletEventCommand and ProcessCoinEventCommand names and you told me purpose of these commands, but one day later I can't remember they meaning and difference :) and they names don't help me with it. Please, give them more meaningful names.

Reply:
Looks like this handler not publishes any events. That's right?

Reply:
fixed.
It publishes

Reply:
cashin only